ext/mbstring/php_mbregex.c

```diff
@@ -1025,15 +1027,9 @@ static void _php_mb_regex_ereg_replace_exec(INTERNAL_FUNCTION_PARAMETERS, OnigOp
 
 	const mbfl_encoding *enc;
 
-	{
-		const char *current_enc_name;
-		current_enc_name = php_mb_regex_get_mbctype();
-		if (current_enc_name == NULL ||
-			(enc = mbfl_name2encoding(current_enc_name)) == NULL) {
-			php_error_docref(NULL, E_WARNING, "Unknown error");
-			RETURN_FALSE;
-		}
-	}
+	enc = mbfl_name2encoding(php_mb_regex_get_mbctype());
```

Comment:
Side note: We'll probably want to cache the mbfl_encoding for the current mbctype, as mbfl_name2encoding() is quite expensive function.